### ISSUE TITLE
Crash when upgrading to v3.1 [thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion `0' failed.]

### ISSUE THREAD

**Issue by acedogblast** [godotengine/godot#28625]
Godot version: Crashes with v3.1 and v3.1.1 works with v3.0.6
OS/device including version: Linux/X11
Issue description:
When upgrading my project from Godot 3.0.6 to 3.1 or 3.1.1 the editor crashes right when it finishes re-importing assets. These files are also animations.
Console output:
```
ERROR: _png_error_function: Not a PNG file
   At: drivers/png/image_loader_png.cpp:62.
libpng error: Not a PNG file
ERROR: _load_image: PNG Corrupted
   At: drivers/png/image_loader_png.cpp:98.
ERROR: _reimport_file: Error importing: res://Graphics/Pictures/dialup.png
   At: editor/editor_file_system.cpp:1618.
ERROR: _png_error_function: Not a PNG file
   At: drivers/png/image_loader_png.cpp:62.
libpng error: Not a PNG file
ERROR: _load_image: PNG Corrupted
   At: drivers/png/image_loader_png.cpp:98.
ERROR: _reimport_file: Error importing: res://Graphics/Transitions/RotatingPieces.png
   At: editor/editor_file_system.cpp:1618.
Godot_v3.1.1-stable_x11.64: thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion `0' failed.
```

**Comment by acedogblast** [comment]
I removed the files but godot still crashes on loading the project. Now only gives the last error messange:
`Godot_v3.1.1-stable_x11.64: thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion `0' failed.`

**Comment by acedogblast** [comment]
When deleting the 2 mentioned files before updating to v3.1.1, I get some new errors:
`WARNING: _png_warn_function: Interlace handling should be turned on when using png_read_image
   At: drivers/png/image_loader_png.cpp:71.
WARNING: _png_warn_function: Interlace handling should be turned on when using png_read_image
   At: drivers/png/image_loader_png.cpp:71.
WARNING: _png_warn_function: bKGD: invalid index
   At: drivers/png/image_loader_png.cpp:71.
WARNING: _png_warn_function: bKGD: invalid index
   At: drivers/png/image_loader_png.cpp:71.
ERROR: create: Index p_height - 1=21255 out of size (MAX_HEIGHT=16384)
   At: core/image.cpp:1461.
libpng warning: Image width is zero in IHDR
libpng warning: Image height is zero in IHDR
libpng error: Invalid IHDR data
ERROR: _lossless_pack_png: Method/Function Failed, returning: PoolVector<uint8_t>()
   At: drivers/png/image_loader_png.cpp:312.
ERROR: create: Index p_height - 1=21255 out of size (MAX_HEIGHT=16384)
   At: core/image.cpp:1461.
libpng warning: Image width is zero in IHDR
libpng warning: Image height is zero in IHDR
libpng error: Invalid IHDR data
ERROR: _lossless_pack_png: Method/Function Failed, returning: PoolVector<uint8_t>()
   At: drivers/png/image_loader_png.cpp:312.
WARNING: _png_warn_function: Interlace handling should be turned on when using png_read_image
   At: drivers/png/image_loader_png.cpp:71.
Godot_v3.1.1-stable_x11.64: thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion `0' failed.
`

**Comment by volzhs** [comment]
can you share the png file?

**Comment by acedogblast** [comment]
Just rechecked one of the files are corrupted and I have removed it. The other still works.
![dialup](https://user-images.githubusercontent.com/34040430/57108228-16b68a00-6d00-11e9-9932-4656f5a597b5.png)

**Comment by acedogblast** [comment]
But I have removed both before and Godot still crashes without them.

**Comment by volzhs** [comment]
I found `GIF89a` in your file.
it's actually gif file. I really surprised it worked with 3.0.x

**Comment by acedogblast** [comment]
I didn't use that file yet. I was just part of the project. Could this be fixed without needed for me to redo my whole project if I want to update to use future Godot versions.

**Comment by acedogblast** [comment]
I have a new console output. This time I deleted and replaced all asset files with their corresponding .import files.
WARNING: _png_warn_function: bKGD: invalid index
   At: drivers/png/image_loader_png.cpp:71.
WARNING: _png_warn_function: bKGD: invalid index
   At: drivers/png/image_loader_png.cpp:71.
ERROR: create: Index p_height - 1=21255 out of size (MAX_HEIGHT=16384)
   At: core/image.cpp:1461.
libpng warning: Image width is zero in IHDR
libpng warning: Image height is zero in IHDR
libpng error: Invalid IHDR data
ERROR: _lossless_pack_png: Method/Function Failed, returning: PoolVector<uint8_t>()
   At: drivers/png/image_loader_png.cpp:312.
ERROR: create: Index p_height - 1=21255 out of size (MAX_HEIGHT=16384)
   At: core/image.cpp:1461.
libpng warning: Image width is zero in IHDR
libpng warning: Image height is zero in IHDR
libpng error: Invalid IHDR data
ERROR: _lossless_pack_png: Method/Function Failed, returning: PoolVector<uint8_t>()
   At: drivers/png/image_loader_png.cpp:312.
WARNING: _png_warn_function: Interlace handling should be turned on when using png_read_image
   At: drivers/png/image_loader_png.cpp:71.
WARNING: _png_warn_function: Interlace handling should be turned on when using png_read_image
   At: drivers/png/image_loader_png.cpp:71.
WARNING: _png_warn_function: Interlace handling should be turned on when using png_read_image
   At: drivers/png/image_loader_png.cpp:71.
ERROR: import: Format not supported for WAVE file (not PCM). Save WAVE files as uncompressed PCM instea
   At: editor/import/resource_importer_wav.cpp:163.
ERROR: _reimport_file: Error importing: res://Audio/SE/computeropen.WAV
   At: editor/editor_file_system.cpp:1618.
ERROR: import: Format not supported for WAVE file (not PCM). Save WAVE files as uncompressed PCM instea
   At: editor/import/resource_importer_wav.cpp:163.
ERROR: _reimport_file: Error importing: res://Audio/SE/computerclose.WAV
   At: editor/editor_file_system.cpp:1618.
Godot_v3.1.1-stable_x11.64: thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion `0' failed.

Not sure why it complains about some of my wav files now instead from v3.0.6.

**Comment by acedogblast** [comment]
When running the console I noticed that when the last line is printed is right when the editor crashes.

"Godot_v3.1.1-stable_x11.64: thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion `0' failed."

Looks like something to do with vorbis audio code.

**Comment by acedogblast** [comment]
Ok I believe I am getting to something important. In my project I have 2 PNG files with resolution of 256x21256. This explains the error code`ERROR: create: Index p_height - 1=21255 out of size (MAX_HEIGHT=16384)`

I have opened up these files into a new blank project and got more console output from them:
`ERROR: create: Index p_height - 1=21255 out of size (MAX_HEIGHT=16384)
   At: core/image.cpp:1461.
libpng warning: Image width is zero in IHDR
libpng warning: Image height is zero in IHDR
libpng error: Invalid IHDR data
ERROR: _lossless_pack_png: Method/Function Failed, returning: PoolVector<uint8_t>()
   At: drivers/png/image_loader_png.cpp:312.
ERROR: create: Index p_height - 1=21255 out of size (MAX_HEIGHT=16384)
   At: core/image.cpp:1461.
libpng warning: Image width is zero in IHDR
libpng warning: Image height is zero in IHDR
libpng error: Invalid IHDR data
ERROR: _lossless_pack_png: Method/Function Failed, returning: PoolVector<uint8_t>()
   At: drivers/png/image_loader_png.cpp:312.
ERROR: _lossless_unpack_png: Condition ' len < 4 ' is true. returned: Ref<Image>()
   At: drivers/png/image_loader_png.cpp:261.
ERROR: _load_data: Condition ' img.is_null() || img->empty() ' is true. returned: ERR_FILE_CORRUPT
   At: scene/resources/texture.cpp:594.
ERROR: _load: Failed loading resource: res://.import/Outside(new).png-a20fe2ac9d92768caa8b717375a02408.stex
   At: core/io/resource_loader.cpp:285.
ERROR: _load: Failed loading resource: res://Graphics/Tilesets/Outside(new).png
   At: core/io/resource_loader.cpp:285.
ERROR: _lossless_unpack_png: Condition ' len < 4 ' is true. returned: Ref<Image>()
   At: drivers/png/image_loader_png.cpp:261.
ERROR: _load_data: Condition ' img.is_null() || img->empty() ' is true. returned: ERR_FILE_CORRUPT
   At: scene/resources/texture.cpp:594.
ERROR: _load: Failed loading resource: res://.import/Outside.png-f2e03dc05a677d9fc38028065702448a.stex
   At: core/io/resource_loader.cpp:285.
ERROR: _load: Failed loading resource: res://Graphics/Tilesets/Outside.png
   At: core/io/resource_loader.cpp:285.`

Does Godot have issues with PNG files this large? The interesting thing is that the editor did not crash with the error: `Godot_v3.1.1-stable_x11.64: thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion `0' failed.` but just failed to use them. There is no limit in the PNG specifications according to this post in stackoverflow (https://stackoverflow.com/questions/4109447/file-format-limits-in-pixel-size-for-png-images). Can Godot use larger PNG files?

**Comment by acedogblast** [comment]
Tested with v3.0.6. Godot also can't use the large png file.

**Comment by acedogblast** [comment]
Never mind. I thought I could just crop the large PNG into 2 but first tested to upgrade without the large files to see if it could work. It did not as at the end of re-importing the assets it crashed with the error:
`Godot_v3.1.1-stable_x11.64: thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion `0' failed.`
Changing title to that error.

**Comment by clayjohn** [comment]
Can you post a minimal reproducible example so that other users can reproduce the error and figure it out?

Your explanation has been a little confusing so far. You say the issue seems to be with a png file, but the error you post has to do with audio.

**Comment by acedogblast** [comment]
@clayjohn Sorry for the confusion. Let me to clarify then. My goal is to upgrade a project that I have been working on from v3.0.6 to v3.1 (now v3.1.1). When upgrading the project, Godot re-imports all the assets. When the loading bar for that process completes Godot crashes immediately. I originally assumed that this was a bug in Godot but as volzhs said there where issues in my assets. I temporally removed the assets that were causing issues and I thought that was the reason why Godot crashed. However, they were not the reason (Though I will make a separate issue for support on large PNG files). The actual problem now is the last error that gets printed out to the console, `Godot_v3.1.1-stable_x11.64: thirdparty/misc/stb_vorbis.c:3050: get_window: Assertion 0 failed.`, as I noticed that is given right when Godot crashes hence the current Title of this issue.

I will try to make a minimal reproducible example sometime this afternoon. Thank you for your time to help me with this bug.

**Comment by acedogblast** [comment]
@clayjohn I have found the issue! The problem is not the upgrading process but there was one specific ogg sound file that is causing Godot to crash when re-importing it. I'm am not sure why this file is causing the assertion to fail. It opens and plays just fine in other media players and I was able to remux it in ffmpeg and then it works without crashing. Here is the file and instructions on how to reproduce:
Ogg file: https://drive.google.com/file/d/10ZFW953aArS3JAiqlqmuoQbrYDsokwzX/view?usp=sharing
Create new project in Godot v3.1.1 and simply add the ogg file to the project.
